pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -74,6 +72,10 @@ func updateAvailable(ac v1alpha1.Condition, prometheusConditions []monv1.Prometh
 		return ac
 	}
 
+	if prometheusAvailable.ObservedGeneration != generation {
```

Comment:
hmm if I understand correctly, we want to check the Prometheus condition's  observedGeneration against the Prometheus object's generation.

Reply:
Well I've discussed with Nico and maybe I still don't fully get it, but my understanding is following:
- MonitoringStack spec is updated -> generation is bumped
- OBO is reflecting the change in the Prometheus

Reply:
You have to compare Prometheus.metadata.generation with `Prometheus.status.condition[].observedGeneration` and
not `MonitoringStack.metadata.generation` with `Prometheus.status.condition[].observedGeneration`

Under perfect conditions, it would not actually matter, but if anyone with UPDATE permission to Prometheus is updating Prometheus directly you will end up with:
`MonitoringStack.metadata.generation = 1` and `Prometheus.metadata.generation = 3`

![image (8)](https://user-images.githubusercontent.com/7278896/195624045-a51cc459-fcec-4be5-9156-b1203d45ac11.png)

Reply:
OK thanks. I will update it. And the case (I tried to describe above) can't happen (maybe not...)?